MUXSDKStats/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -581,10 +589,37 @@ - (MUXSDKPlayerData *)getPlayerData {
                                                                     encoding:NSUTF8StringEncoding]];
         }
     }
+
+    // Only report sampling data if the playback is live
+    if (_playbackIsLivestream) {
+        // Sampling Data
+        NSTimeInterval currentDate = [_player.currentItem.currentDate timeIntervalSince1970];
+        playerData.playerProgramTime = [NSNumber numberWithLongLong: (long long)(currentDate * 1000)];
+
+
+        if ([_player.currentItem.seekableTimeRanges count] > 0) {
+            // seekableTimeRanges is ordered, so we only need to look at the last one
+            // Note about seekableTimeRanges: the meaning of the values in seekableTimeRanges appears to change
+            // across OS versions and device types. Sometimes the duration appears to take holdbacks from the HLS
+            // manifest into consideration, and other times not. Use this value with caution.
+            CMTimeRange seekableRange = [_player.currentItem.seekableTimeRanges.lastObject CMTimeRangeValue];
+            CGFloat start = CMTimeGetSeconds(seekableRange.start);
+            CGFloat duration = CMTimeGetSeconds(seekableRange.duration);
+            CGFloat timeOfVideo = CMTimeGetSeconds([[_player currentItem] currentTime]);
+            CGFloat livePosition = start + duration;
+            NSTimeInterval viewStartDate = currentDate - timeOfVideo;
+            NSTimeInterval newestProgramDate = viewStartDate + livePosition;
+
+            playerData.playerLiveEdgeProgramTime = [NSNumber numberWithLongLong:(long long)(newestProgramDate * 1000)];
```

Comment:
I think I'm slightly confused about this calculation, perhaps because we are adding CGFloats to NSDates without using the `addingTimeInterval` API? It might return the same result but having trouble parsing this. This is how I was calculating it:

```
let start = timeRange.start.seconds
let totalDuration = timeRange.duration.seconds
let currentTime = curPlayer.currentTime().seconds
let liveEdge = start + totalDuration
let secondsBehindLive = currentTime - liveEdge
print("  [LATENCY METRIC] Live Edge PDT: ", curDate.addingTimeInterval(-1*secondsBehindLive))
```

Reply:
the calculation here is:
- find the timestamp that represents the beginning of the video ([currentItem currentDate] - timeOfVideo). Our understanding is that currentDate is calculated using the X-PROGRAM-DATE from the manifest
- Add the number of seconds that seekableTimeRanges tells us are available for seeking (livePosition) to that timestamp to calculate the timestamp for the video "live edge"
This calculation is only using values from the video itself

The calculation you have is assuming a relationship between the device date (assuming that is what curDate means) and the video dates which I'm not sure will be true